frontend/src/components/pages/profile/Profile.tsx

```diff
@@ -1,10 +1,186 @@
+import FormControl from "@material-ui/core/FormControl";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import Input from "@material-ui/core/Input";
+import InputLabel from "@material-ui/core/InputLabel";
+import { makeStyles } from "@material-ui/core/styles";
+import OutlinedInput from "@material-ui/core/OutlinedInput";
+import { PersonProps } from "../../../services/mock_profile_backend";
+import { USER_ID } from "../../../utils/temporary_testing_consts";
+import { ProfileHandlerService } from "../../../services/profile_handler_service";
+import { MockProfileBackendService } from "../../../services/mock_profile_backend";
+import { ServiceContext } from "../../contexts/contexts";
+import { ServiceHandlers } from "../../contexts/contexts";
+import { Theme } from "@material-ui/core/styles";
+import TextField from '@material-ui/core/TextField';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      display: 'flex',
+      flexWrap: 'wrap',
+      marginTop: '100px',
+      marginLeft: theme.spacing(4),
+    },
+    '& input:invalid + fieldset': {
+      borderColor: 'red',
+      borderWidth: 2,
+    },
+    textField: {
+      marginLeft: theme.spacing(1),
+      marginRight: theme.spacing(1),
+      width: '25ch',
+    },
+  }),
+);
+
+interface ProfileProps {
+  profileHandlerService: ProfileHandlerService;
+}
+
+interface ProfileState {
+  personProps: PersonProps;
+}
+
+export default function Profile(props: ProfileProps, state: ProfileState) {
+
+  const classes = useStyles();
+
+  /** 
+   * ServiceHandlers is an object containing various TS Handlers and provides
+   * functionality to communicate data from the frontend to the backend
+   */
+  const serviceHandlers = React.useContext(ServiceContext);
+  const profileHandlerService = serviceHandlers.profileHandlerService;
+  
+  const [person, setPerson] = React.useState<PersonProps|undefined>(undefined);
+  const [profileId, setProfileId] = React.useState<string>(undefined);
+  const [submitSuccess, setSubmitSuccess] = React.useState<Boolean|undefined>(undefined);
+  
+  React.useEffect(() => {
+    (async() => {
+      const personPromise = await profileHandlerService.getPerson(USER_ID);
+      setPerson(personPromise);
+    })();
+  }, [profileId]); /** Re-renders Profile only when the userId changes */
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPerson({...person, email: event.target.value});
+  }
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPerson({...person, nickname: event.target.value});
+  }
+
+  const handlePronounsChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPerson({...person, pronouns: event.target.value});
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    (async() => {
+      const success = (person.email === "" || person.nickname === "")
```

Comment:
This is clever but I think you'll actually want to break it out into a separate statement so you can give a good error message (eg. `Please set a nickname before submitting`)